pal/pacer/pacer-web.ts

```diff
@@ -96,32 +97,41 @@ export class Pacer {
         this._stHandle = this._stTime(updateCallback);
 
         this._isPlaying = true;
+        this._frameCount = 0;
     }
 
     stop (): void {
         if (!this._isPlaying) return;
         this._ctTime(this._stHandle);
         this._stHandle = 0;
         this._isPlaying = false;
+        this._frameCount = 0;
     }
 
     _handleRAF = () => {
-        if (performance.now() - this._start < this._delay) {
+        const elapseTime = performance.now() - this._startTime;
+        const elapseFrame = Math.floor(elapseTime / this._frameTime);
+        if (elapseFrame < this._frameCount) {
             this._rAF.call(window, this._handleRAF);
-        } else if (this._callback) {
-            this._callback();
+        } else {
+            this._frameCount++;
+            if (this._callback) {
+                this._callback();
+            }
+        }
+        if (performance.now() - this._startTime > FRAME_RESET_TIME) {
+            this._startTime = performance.now();
+            this._frameCount = 0;
         }
     };
 
     private _stTime (callback: () => void) {
-        const currTime = performance.now();
-        const elapseTime = Math.max(0, (currTime - this._startTime));
-        const timeToCall = Math.max(0, this._frameTime - elapseTime);
         if (EDITOR || this._rAF === undefined || globalThis.__globalXR?.isWebXR) {
+            const currTime = performance.now();
+            const elapseTime = Math.max(0, currTime - this._startTime);
+            const timeToCall = Math.max(0, this._frameTime - elapseTime);
             return setTimeout(callback, timeToCall);
```

Comment:
As we are using the previous code only for setTimeout branch, I think it's safe to use directly `this._frameTime` here to setTimeout, no need to do any timeToCall calculations